File: 3659. Partition Array Into K-Distinct Groups/Solution.cpp
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    bool partitionArray(vector<int>& nums, int k)
    {
        if (nums.size() % k != 0)
            return false;
        int groups = nums.size() / k;
        unordered_map<int, int> cnt;
        for (int x : nums)
            if (++cnt[x] > groups)
                return false;
        return true;
    }
};

File: 3659. Partition Array Into K-Distinct Groups/Solution.py
from collections import defaultdict


class Solution:
    def partitionArray(self, nums: list[int], k: int) -> bool:
        nums_len = len(nums)
        if nums_len % k != 0:
            return False
        groups = nums_len // k
        cnt = defaultdict(int)
        for x in nums:
            cnt[x] += 1
            if cnt[x] > groups:
                return False
        return True
